src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -250,7 +461,9 @@ private void SetResolverState(int submissionId, SdkResolver resolver, object sta
             {
                 ConcurrentDictionary<SdkResolver, object> resolverState = _resolverStateBySubmission.GetOrAdd(
                     submissionId,
-                    _ => new ConcurrentDictionary<SdkResolver, object>(NativeMethodsShared.GetLogicalCoreCount(), _resolvers.Count));
+                    _ => new ConcurrentDictionary<SdkResolver, object>(
+                        NativeMethodsShared.GetLogicalCoreCount(),
+                        ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_4) ? _specificResolversManifestsRegistry.Count + _generalResolversManifestsRegistry.Count : _resolversList.Count));
```

Comment:
About this line I am not sure I would like it this way. Previously at this moment we knew the total number of resolvers and could set the dictionary well. Now we do not know the number of resolvers, only number of manifests, each could in theory have >1 resolver. I wonder that it might be better not to specify the capacity at all than specify and underestimate it.